Mathlib/Order/Heyting/Basic.lean

```diff
@@ -442,6 +442,9 @@ theorem le_himp_himp : a ≤ (a ⇨ b) ⇨ b :=
   le_himp_iff.2 inf_himp_le
 #align le_himp_himp le_himp_himp
 
+@[simp] lemma himp_eq_himp : b ⇨ a = a ⇨ b ↔ a = b := by simp [le_antisymm_iff]
+lemma himp_ne_himp : b ⇨ a ≠ a ⇨ b ↔ a ≠ b := himp_eq_himp.not
```

Comment:
I'd be tempted to put `_iff`s on these lemmas, just to make their statement more obvious